hips/tiles/fetch.py

```diff
@@ -0,0 +1,112 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import numpy as np
+import urllib.request
+import concurrent.futures
+from typing import Generator
+from ..tiles import HipsSurveyProperties, HipsTile, HipsTileMeta
+
+__all__ = [
+    'HipsTileFetcher',
+]
+
+
+class HipsTileFetcher:
+    """Fetch a list of HiPS tiles.
+
+    Parameters
+    ----------
+    tile_indices : np.ndarray
+        List of index values for HiPS tiles
+    hips_order : int
+        Order of the HiPS survey
+    hips_survey : `~hips.HipsSurveyProperties`
+        HiPS survey properties
+    tile_format : {'fits', 'jpg', 'png'}
+        Format of HiPS tile
+    progress_bar : bool
+        Show a progress bar for tile fetching and drawing
+    """
+
+    def __init__(self, tile_indices: np.ndarray, hips_order: int, hips_survey: HipsSurveyProperties,
+                 tile_format: str, progress_bar: bool) -> None:
+        self.tile_indices = tile_indices
+        self.hips_order = hips_order
+        self.hips_survey = hips_survey
+        self.tile_format = tile_format
+        self.progress_bar = progress_bar
+
+    def fetch_tile_threaded(self, url: str) -> Generator:
+        """Fetch a HiPS tile asynchronously."""
+        with urllib.request.urlopen(url, timeout=60) as conn:
+            return conn.read()
+
+    @property
+    def tiles(self) -> np.ndarray:
+        """Generator function to fetch HiPS tiles from a remote URL."""
+        tile_urls, tile_metas = [], []
+        for healpix_pixel_index in self.tile_indices:
+            tile_meta = HipsTileMeta(
+                order=self.hips_order,
+                ipix=healpix_pixel_index,
+                frame=self.hips_survey.astropy_frame,
+                file_format=self.tile_format,
+            )
+            tile_urls.append(self.hips_survey.tile_url(tile_meta))
+            tile_metas.append(tile_meta)
+
+        raw_responses = []
+        with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
```

Comment:
You shouldn't hard-code `max_workers=5`  here, but somehow give users of this class the option to control this parameter.